File: backend/app/schemas/lineage.py
"""Lineage schemas for the technology genealogy graph."""
from pydantic import BaseModel, ConfigDict


class LineageNode(BaseModel):
    id: int  # Item ID
    title: str
    source: str
    priority: str | None = None
    llm_score: int = 0
    year: int | None = None
    url: str


class LineageEdgeRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    parent_id: int
    child_id: int
    relationship_type: str


class LineageGraph(BaseModel):
    center_id: int | None = None
    nodes: list[LineageNode]
    edges: list[LineageEdgeRead]

File: frontend/src/api/lineage.ts
import { apiGet } from "./client";

export type LineageNode = {
  id: number;
  title: string;
  source: string;
  priority: string | null;
  llm_score: number;
  year: number | null;
  url: string;
};

export type LineageEdge = {
  parent_id: number;
  child_id: number;
  relationship_type: string;
};

export type LineageGraph = {
  center_id: number | null;
  nodes: LineageNode[];
  edges: LineageEdge[];
};

export const fetchItemLineage = (itemId: number, depth = 2) =>
  apiGet<LineageGraph>(`/lineage/item/${itemId}?depth=${depth}`);

export const fetchCategoryLineage = (slug: string) =>
  apiGet<LineageGraph>(`/lineage/category/${slug}`);
